pkg/cloud/services/compute/instance.go

```diff
@@ -580,6 +582,20 @@ func (s *Service) InstanceDelete(machine *clusterv1.Machine, openStackMachine *i
 		return err
 	}
 
+	err = util.PollImmediate(RetryIntervalInstanceStatus, TimeoutInstanceDelete, func() (bool, error) {
+		_, err = s.GetInstance(parsed.ID())
+		if err != nil {
+			if capoerrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("error deleting Openstack instance %s, %v", parsed.ID(), err)
```

Comment:
better to update like Line 579-580:

```go
record.Warnf(openStackMachine, "FailedDeleteServer", "Failed to deleted server %s: %v", parsed.ID(), err)
return err
```

Reply:
We need to add the same logic for deleting bastion.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/services/compute/bastion.go#L34-L38

Reply:
I think we should phrase the message a little bit differently to indicate the wait timeout so we now which one of the cases we hit.

Reply:
@sbueringer does the error report already contains in Poll logic? I checked all other Poll method does not have special handle

Reply:
Ah probably there is something in the poll timeout error, you're right this should be good enough